src/open-api/from-open-api.ts

```diff
@@ -21,7 +21,7 @@ const supportedHttpMethods = Object.keys(
 export async function fromOpenApi(
   document: string | OpenAPI.Document | OpenAPIV3.Document | OpenAPIV2.Document,
```

Comment:
Just noticed the `string` type here.

What should happen when we pass in a `string`? Should we support `yaml` or only `json` format?

Currently, my dereference function only supports `objects`.

Reply:
Thank you for your pull request :)

Yeah, good point. I think we probably want to support both `yaml` and `json`? I thought one of those `yaml` packages support both `json` and `yaml` but I could be imagining that.

Reply:
You're right! The [yaml](https://www.npmjs.com/package/yaml) package support both JSON and YAML syntax, so we don't need to handle that logic ourselves!

I added the library and some test cases.